apis/projectcontour/v1/httpproxy.go

```diff
@@ -411,6 +411,12 @@ type RateLimitDescriptorEntry struct {
 	// +optional
 	RequestHeader *RequestHeaderDescriptor `json:"requestHeader,omitempty"`
 
+	// RequestHeaderValueMatch defines a descriptor entry that's populated
+	// if the request's headers match a set of 1+ match criteria. The
+	// descriptor key is "header_match", and the descriptor value is static.
+	// +optional
+	RequestHeaderValueMatch *RequestHeaderValueMatchDescriptor `json:"requestHeaderValueMatch,omitempty"`
```

Comment:
We probably can't remove it but this sorta overlaps with the existing `RequestHeader` a bit since the `HeaderMatchCondition` type has a `Present` field. Difference I guess is that this one will set the descriptor key and value to be static. Might be good to call out in some docs

Reply:
Yeah, it's similar but I think different enough to be its own thing (as it is in Envoy). I'll do a follow-up PR to add some more docs on each descriptor entry type to the rate limiting page; the API godoc covers it somewhat but some additional material would be helpful too.

Reply:
#3604